api/v1alpha1/groupversion_info.go

```diff
@@ -20,17 +20,48 @@ limitations under the License.
 package v1alpha1
 
 import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/runtime/schema"
-	"sigs.k8s.io/controller-runtime/pkg/scheme"
 )
 
 var (
 	// GroupVersion is group version used to register these objects
 	GroupVersion = schema.GroupVersion{Group: "ocs.openshift.io", Version: "v1alpha1"}
 
 	// SchemeBuilder is used to add go types to the GroupVersionKind scheme
-	SchemeBuilder = &scheme.Builder{GroupVersion: GroupVersion}
+	SchemeBuilder = &Builder{GroupVersion: GroupVersion}
 
 	// AddToScheme adds the types in this group-version to the given scheme.
 	AddToScheme = SchemeBuilder.AddToScheme
 )
+
+// copied from https://github.com/kubernetes-sigs/controller-runtime/blob/2eb879/pkg/scheme/scheme.go
+// Copyright 2018 The Kubernetes Authors.
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// gist, controller-runtime scheme allows for gradual registration of types vs kube runtime which is more suited
+// for registering all types during construction of the builder in addition to providing gradual registration which is
+// used by controller-runtime and required parts are copied as below.
+
+// +kubebuilder:object:generate=false
+// Builder builds a new Scheme for mapping go types to Kubernetes GroupVersionKinds.
```

Comment:
1. We should not have copywrite comment in the middle of a file.
2. I don't really understand the need for this change

Reply:
1. the code is copyrighted by Kubernetes and licensed w/ apache and so just preserving that note here, I could move this to a separate file though to be clean
2. either this or something similar to https://github.com/ceph/ceph-csi-operator/pull/105 be done, I chose the former as not to take away the flexibility of registering API types in their individual files